mysql-test/suite/innodb/t/shrink_cached_undo.test

```diff
@@ -0,0 +1,43 @@
+--source include/have_innodb.inc
+--source include/have_sequence.inc
+--source include/have_debug.inc
+--source include/not_embedded.inc
+--echo #
+--echo # MDEV-35398 Improve shrinking of system tablespace
+--echo #
+# Make pre-10.6.13 version data directory which skips the
+# purging of cached undo log records
+let $restart_parameters=--debug_dbug=d,skip_cached_undo;
+--source include/restart_mysqld.inc
+set global innodb_file_per_table= 0;
+create table t1(f1 int not null)engine=innodb;
+create table t2(f2 int not null)engine=innodb;
+INSERT INTO t1 SELECT seq FROM seq_1_to_32768;
+INSERT INTO t2 SELECT seq FROM seq_1_to_32768;
```

Comment:
Can these be any smaller? Can we prepend each of these with
```sql
SET STATEMENT unique_checks=0,foreign_key_checks=0 FOR
```
so that there will be less undo log written? Or is this undo log important for the test?

Moreover, since we are using debug instrumentation anyway, could we use `SET GLOBAL innodb_limit_optimistic_insert_debug` to bloat the tables while inserting much fewer rows?

Reply:
For some reason, it is not repeating with `innodb_limit_optimistic_insert_debug`. But I added `include/big_test.inc`